File: tracks/python/leap.py
"""
Leap Year Calculation Module

This module provides functionality to determine whether a given year is a leap year.
A leap year is a year that is divisible by 4, except for end-of-century years, 
which must be divisible by 400. The module contains the following function:

- leap_year(year): Determines if the specified year is a leap year.
"""


def leap_year(year):
    """Determine whether a specified year is a leap year.

    :params year: int - year to be checked.
    :return: bool - True if the year is a leap year, False otherwise.
    """
    return (year % 4 == 0) and (year % 100 != 0 or year % 400 == 0)

File: tracks/javascript/leap.js

/**
 * Determines if a given year is a leap year.
 * A year is a leap year if it is divisible by 4 but not by 100, except for
 * years that are divisible by 400.
 * @param {int} year the year to be identified
 * @returns True if a year is leap year, False otherwise
 */
export const isLeap = year => {
    return (year % 4 == 0) && (year % 100 != 0 || year % 400 == 0);
};
  